runsc/container/container.go

```diff
@@ -1027,13 +1069,23 @@ func (c *Container) createGoferFilestoreInDir(ovlConf config.Overlay2) (*os.File
 	// This file will be deleted when the container exits.
 	filestoreFile, err := os.CreateTemp(filestoreDir, "runsc-filestore-")
 	if err != nil {
-		return nil, boot.VanillaGofer, fmt.Errorf("failed to create a temporary file inside %q: %v", filestoreDir, err)
+		return nil, boot.GoferMountConf{}, fmt.Errorf("failed to create a temporary file inside %q: %v", filestoreDir, err)
 	}
 	if err := unix.Unlink(filestoreFile.Name()); err != nil {
-		return nil, boot.VanillaGofer, fmt.Errorf("failed to unlink temporary file %q: %v", filestoreFile.Name(), err)
+		return nil, boot.GoferMountConf{}, fmt.Errorf("failed to unlink temporary file %q: %v", filestoreFile.Name(), err)
 	}
 	log.Debugf("Created an unnamed filestore file at %q", filestoreDir)
-	return filestoreFile, boot.AnonOverlay, nil
+	return filestoreFile, boot.GoferMountConf{Lower: boot.Lisafs, Upper: boot.AnonOverlay}, nil
+}
+
+// shouldSpawnGofer returns true if we need to spawn the gofer process.
+func (c *Container) shouldSpawnGofer() bool {
```

Comment:
We should not define this method on `Container`. Because in practice, it is being called on partially initialized `Container` structs (i.e. c.GoferMountConfs needs to be initialized, but c.Sandbox may still be nil).

I think it is better for it to be an anonymous function and take in `[]boot.GoferMountConfs`. I also plan to call this from runsc/boot/loader.go, when starting the gofer process monitor later (for sanity check that at least 1 monitor was started if the gofer process was spawned).

Reply:
Makes sense. Will do.